WordPress/gutenberg
# Button, Icon: fix iconSize prop doesn't work with some icons

Fix #43745

## What?
~~This PR fixes an issue where the `iconSize` prop of the `Button` component was not reflected in some icons.~~
This PR fixes a problem in the `@wordpress/components` package where the `iconSize` prop of the `Button` and `Icon` component was not reflected in some icons.


## Why?
This is because the `iconSize` prop is overridden if the icon's svg element has a `width` / `height` attribute, as noted in [this comment](https://github.com/WordPress/gutenberg/pull/43747#pullrequestreview-1093433535).

The following are those icons:

- [atSymbol](https://github.com/WordPress/gutenberg/blob/4055380fed140501e7f675fb1c10cdf0496917ed/packages/icons/src/library/at-symbol.js)
- [commentEditLink](https://github.com/WordPress/gutenberg/blob/4055380fed140501e7f675fb1c10cdf0496917ed/packages/icons/src/library/comment-edit-link.js)
- [commentReplyLink](https://github.com/WordPress/gutenberg/blob/4055380fed140501e7f675fb1c10cdf0496917ed/packages/icons/src/library/comment-reply-link.js)
- [dragHandle](https://github.com/WordPress/gutenberg/blob/4055380fed140501e7f675fb1c10cdf0496917ed/packages/icons/src/library/drag-handle.js)
- [helpFilled](https://github.com/WordPress/gutenberg/blob/4055380fed140501e7f675fb1c10cdf0496917ed/packages/icons/src/library/help-filled.js)

## How?

In the internal Icon component, I have made the passed props (such as iconSize) take precedence over the attributes of the SVG element.

## Testing Instructions
Rewrite edit.js for any of the core blocks as follows and confirm that they all render at the expected size:

_@add:_ This example also uses `@wordpress/icons` Icon, which is incorrect. It's not a problem with the `@wordpress/icons` package, as [this comment states](https://github.com/WordPress/gutenberg/pull/43821#pullrequestreview-1096509151).

```javascript
/**
 * WordPress dependencies
 */
import { useBlockProps } from '@wordpress/block-editor';
import { Button } from '@wordpress/components';
import { Icon, help, helpFilled } from '@wordpress/icons';

export default function CodeEdit() {
	const blockProps = useBlockProps();
	return (
		<div { ...blockProps }>
			<p>Icon without `size` prop (24px expected)</p>
			<Icon icon={ help } />
			<Icon icon={ helpFilled } />

			<p>Icon with `size` prop (60px expected)</p>
			<Icon icon={ help } size="60" />
			<Icon icon={ helpFilled } size="60" />

			<p>Button without `iconSize` prop (24px expected)</p>
			<Button icon={ help } />
			<Button icon={ helpFilled } />

			<p>Button with `iconSize` prop (60px expected)</p>
			<Button icon={ help } iconSize="60" />
			<Button icon={ helpFilled } iconSize="60" />
		</div>
	);
}
```

- The `help` icon on the left has no width / height attributes on the svg element.
- The `helpFilled` icon on the right has width / height attributes on the svg element.

### On trunk

![on-trunk](https://user-images.githubusercontent.com/54422211/188179346-1f61a5a3-7b59-485f-b2ee-9236191da26b.png)

### ON this branch

![on-branch](https://user-images.githubusercontent.com/54422211/188179362-a90141f5-331c-40c7-ad58-d1ef8ea2c443.png)


## Screenshots or screencast
